### PR TITLE
Enable no-copy API in Model API OVMS Adapter to optimize Embedding endpoint

### PR DESCRIPTION
**Pull request by atobiszei**
Related changes:
https://github.com/openvinotoolkit/model_server/pull/2763
https://github.com/openvinotoolkit/model_api/pull/212

Ticket: CVS-155477

### CODE REVIEW

File: mediapipe/calculators/ovms/modelapiovmsadapter.cc [hunk 1]
```diff
@@ -89,14 +89,109 @@ OVMSInferenceAdapter::~OVMSInferenceAdapter() {
...
+        ASSERT_CAPI_STATUS_NULL(OVMS_InferenceRequestAddInput(request, realInputName, inputDataType, inputShape.data(), inputShape.size()));
+        const uint32_t NOT_USED_NUM = 0;
+        ASSERT_CAPI_STATUS_NULL(OVMS_InferenceRequestInputSetData(request,
+            realInputName,
+            reinterpret_cast<void*>(input_tensor.data()),
+            input_tensor.get_byte_size(),
+            OVMS_BUFFERTYPE_CPU,
+            NOT_USED_NUM));
+    }
+    for (const auto& [name, input_tensor] : output) {
```
Comment (dkalinowski) [inline mediapipe/calculators/ovms/modelapiovmsadapter.cc]:
```suggestion
    for (const auto& [name, output_tensor] : output) {
```

File: mediapipe/calculators/ovms/modelapiovmsadapter.cc [hunk 2]
```diff
@@ -89,14 +89,109 @@ OVMSInferenceAdapter::~OVMSInferenceAdapter() {
...
+    size_t dimCount = 42;
+    OVMS_BufferType bufferType = (OVMS_BufferType)199;
+    uint32_t deviceId = 42;
+    const char* outputName{nullptr};
+    for (size_t i = 0; i < outputCount; ++i) {
+        ASSERT_CAPI_STATUS_NULL(OVMS_InferenceResponseOutput(response, i, &outputName, &datatype, &shape, &dimCount, &voutputData, &bytesize, &bufferType, &deviceId));
+        if (std::find(outputsSet.begin(), outputsSet.end(), outputName) == outputsSet.end()) {
+            output.emplace(outputName, std::move(makeOvTensor(datatype, shape, dimCount, voutputData, bytesize)));
+        } else {
+            //output.emplace(outputName, input.at(outputName));
```
Comment (dkalinowski) [inline mediapipe/calculators/ovms/modelapiovmsadapter.cc]: ?

Comment (dkalinowski) [reply mediapipe/calculators/ovms/modelapiovmsadapter.cc]: does it mean that we only get selected outputs?

File: mediapipe/calculators/ovms/modelapiovmsadapter.cc [hunk 3]
```diff
@@ -213,6 +324,21 @@ ov::PartialShape OVMSInferenceAdapter::getInputShape(const std::string& inputNam
     }
     return ovShape;
 }
+ov::PartialShape OVMSInferenceAdapter::getOutputShape(const std::string& inputName) const {
```
Comment (dkalinowski) [inline mediapipe/calculators/ovms/modelapiovmsadapter.cc]:
```suggestion
ov::PartialShape OVMSInferenceAdapter::getOutputShape(const std::string& outputName) const {
```

File: mediapipe/calculators/ovms/modelapiovmsadapter.cc [hunk 4]
```diff
@@ -213,6 +324,21 @@ ov::PartialShape OVMSInferenceAdapter::getInputShape(const std::string& inputNam
...
+    auto it = outShapesMinMaxes.find(inputName);
+    if (it == outShapesMinMaxes.end()) {
+        LOG(INFO) << "Could not find input:" << inputName;
+        throw std::runtime_error(std::string("Adapter could not find input:") + inputName);
```
Comment (dkalinowski) [inline mediapipe/calculators/ovms/modelapiovmsadapter.cc]:
```suggestion
        throw std::runtime_error(std::string("Adapter could not find output:") + outputName);
```

File: mediapipe/calculators/ovms/modelapiovmsadapter.cc [hunk 5]
```diff
@@ -213,6 +324,21 @@ ov::PartialShape OVMSInferenceAdapter::getInputShape(const std::string& inputNam
     }
     return ovShape;
 }
+ov::PartialShape OVMSInferenceAdapter::getOutputShape(const std::string& inputName) const {
+    auto it = outShapesMinMaxes.find(inputName);
```
Comment (dkalinowski) [inline mediapipe/calculators/ovms/modelapiovmsadapter.cc]:
```suggestion
    auto it = outShapesMinMaxes.find(outputName);
```

**Review by dkalinowski — APPROVED** [review]

File: mediapipe/calculators/ovms/modelapiovmsadapter.cc [hunk 6]
```diff
@@ -70,31 +70,109 @@ OVMSInferenceAdapter::~OVMSInferenceAdapter() {
...
+    size_t dimCount = 42;
+    OVMS_BufferType bufferType = (OVMS_BufferType)199;
+    uint32_t deviceId = 42;
+    const char* outputName{nullptr};
+    for (size_t i = 0; i < outputCount; ++i) {
+        ASSERT_CAPI_STATUS_NULL(OVMS_InferenceResponseOutput(response, i, &outputName, &datatype, &shape, &dimCount, &voutputData, &bytesize, &bufferType, &deviceId));
+        if (std::find(outputsSet.begin(), outputsSet.end(), outputName) == outputsSet.end()) {
+            output.emplace(outputName, std::move(makeOvTensor(datatype, shape, dimCount, voutputData, bytesize)));
+        } else {
+            //output.emplace(outputName, input.at(outputName));
```
Comment (michalkulakowski) [inline mediapipe/calculators/ovms/modelapiovmsadapter.cc]: This line should probably be removed

**Review by michalkulakowski — APPROVED** [review]

**Pull request merged**